### DISCUSSION TITLE
[22.02.24] 프론트엔드 회의

### DISCUSSION THREAD

**Discussion by devhyun637** [woowacourse/prolog#700] [frontend]
프론트엔드 회의 때 공유 및 논의할 내용입니다.

### 티케

1. **업무진행상황**을 한눈에 파악하고 싶음.
    - 서로의 스케줄 관리를 한눈에 파악할 수 있는 게 있으면 좋을 것 같음.
    - 큰 리스트를 나누고, 작은 테스크를 가지고 하는 것이 보이면 좋겠는데, github의 projects는 손이 잘 안감.
    - 고민 : 깃으로 계속한다 or 트렐로와 같은 업무 협업 툴을 사용? (이슈 생성은 동일)

2. **코드리뷰** 문화 도입을 해보면 어떨까?
    - 어프로브 문화를 도입하는 것 (approve 1이상이면 머지)
    - 어떤 작업을 했는지, 어떤 파일이 수정되었는지 한눈에 파악하기 위한 용도
    - 빡세게 작업 안하고, 정말 확인하는 용도로만

3. **문서화** 도입
    - 어떤 작업을 했는지에 관해서 작성하는 것이 있으면 좋겠음.
    - PR 메세지 말고 히스토리를 쌓을 수 있는 것이 있다면 좋지 않을까?
    - 깃헙 위키를 계속 관리하는 것도 좋음!

4. 개인적인 일정 계획 공유
    -  ~ 2/28: 기획 및 화면 다시 정리 +  interface, model 설정 / API 문서화가 찾기 (없다면 요구하기)
    -  ~ 3/7: UI 정리하기 (역량 페이지 및 리포트 페이지)
    -  ~ 3/10: 못한거 마무리 + 문서화
    -  ~ 3/17: API 연동 및 테스트(QA)
    - ~ 3/21: 못한거 마무리 + 문서화

**Comment by sunhpark42** [comment]
1. UI Figma로 정리
    - adobe XD 보다 접근성에 좋을 것이라고 생각, 변화를 실시간으로 공유하기 유용함.
    
2. 타입스크립트, 스타일 컨벤션 고정
    - 현재 emotion의 props를 실험적으로 적용해 봤늗데 관련 해서 통일 혹은 이전처럼 스타일 컴포넌트만 사용하는 식으로 변경
    - 타입스크립트 전역 namespace 사용 관련 / 요즘 트렌드는 아니긴 해서 이 부분도 논의해봐도 좋을 것 같습니다.

3. 작업진행상황 공유 수단
    - 이건 프론트 내에서도 공유가 되었으면 하나, 전체 팀적으로도 공유가 되었으면 좋겠음
    - 현재 어떤 이슈가 진행중이고, 담당자가 누구인지 확인이 안되는 이슈
    - 전체적으로 논의 / 컨벤션이 필요할 것 같다.
    
4. 리액트 라우터 v6 업그레이드 관한 건
    - 최신 버전 사용해 보는 것도 좋을 것 같음
    - 다만 useHistory 관련 코드가 전부 사라지고, Path가 상대경로로 전환되어 마이그레이션 비용이 큼.

**Comment by sunhpark42** [comment]
+) 개인적인 일정 계획 공유
    - ~ 2/28 인기있는 학습로그 메인페이지 추가
    - 3월 이후 계획은 아직 없습니다~

**Comment by gracefulBrown** [comment]
## 진행 상황 공유 계획 수립
- 누가 어떤 일을 진행하고 있는지 파악하기 어려움
- 데일리 스레드 부활?
- 주간 회의 진행 부활?

**Comment by gracefulBrown** [comment]
## 기능
- 역량 / 리포트 (티케)
  - https://github.com/woowacourse/prolog/discussions/610
- 메인페이지 (서니)

## 문서화
- api 문서화 (브라운)
- 프론트 문서화: 위키 or TS doc

## UI 도구
- adobe XD -> Figma
- Figma 어떻게 쓰는지 코치들에게 물어보기

## 업무 프로세스
- 주간회의
  - 월요일(?) 8,9시 주간 회의 공유할 내용 있는 사람 참여하기
- 데일리 스레드





- 모바일 대응 (서니)
